flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalRecoveredInputChannel.java

```diff
@@ -42,8 +42,17 @@
 			TaskEventPublisher taskEventPublisher,
 			int initialBackOff,
 			int maxBackoff,
+			int networkBuffersPerChannel,
 			InputChannelMetrics metrics) {
-		super(inputGate, channelIndex, partitionId, initialBackOff, maxBackoff, metrics.getNumBytesInLocalCounter(), metrics.getNumBuffersInLocalCounter());
+		super(
```

Comment:
I'm not sure if I understand this bug and the fix. Why is allocating exclusive buffers for `LocalRecoveredInputChannel` fixing the problem? Isn't it just reducing the window for the live lock to happen? What if downstream tasks are scheduled with a significant delay (exclusive buffers assignment happens after upstream tasks already acquired lot's of buffers).

In other words, Isn't this a semi fix for this bug https://issues.apache.org/jira/browse/FLINK-13203

Reply:
Hm you are right, it doesn't solve it completely after having read the ticket. However, without a solution for FLINK-13203, there will also not be a real solution here.
On the other hand, it's inherently wrong to treat local and remote channels differently during recovery (they even share the same implementation). So this commit is still fixing the issue in a best effort manner and certainly helps to improve build stability, which is an improvement of its own.